static/js/src/PurchaseModal/PurchaseModal.tsx

```diff
@@ -229,6 +229,7 @@ const PurchaseModal = ({
                 product={product}
                 quantity={quantity}
                 action={action}
+                isCardValid={isCardValid}
```

Comment:
The cypress test seems to be failing due the buy button being disabled despite a valid card being added. 

For as much as I understand. `isCardValid` is set to `false` and I cannot see that status changed. And a result the button never gets enabled.

Reply:
`isCardValid` has to check to see if the user has a card already set as part of his customer info and if it exists then have it initialised to true.

At the moment `isCardValid` status get's changed from its default `false` status only if the user edits his details.